crates/node/src/lib.rs

```diff
@@ -302,14 +305,26 @@ impl Node {
                 root_hash,
                 artifact,
             } => {
-                self.handle_state_delta(
-                    source,
-                    context_id,
-                    author_id,
-                    root_hash,
-                    artifact.into_owned(),
-                )
-                .await?;
+                let identities = self.ctx_manager.get_context_owned_identities(context_id)?;
+
+                let Some(our_identity) = identities.into_iter().choose(&mut thread_rng()) else {
+                    bail!("no identities found for context: {}", context_id);
+                };
+
+                let our_sending_key = self
+                    .ctx_manager
+                    .get_own_signing_key(&context_id, &our_identity)?;
```

Comment:
we're not decrypting with "our sender key"

we're decrypting with "their sender key", so you should be passing `author_id` to `get_own_signing_key`

also, that method isn't named right